### ISSUE TITLE
Sitemap:Text [%.1f] Does Not Display Proper Float Number

### ISSUE THREAD

**Issue by lsafelix75** [openhab/openhab-core#428]
On openHAB 2.4.0 Build #1411

LR_Humidity item is defined as Number:Dimensionless

> Text item=LR_Humidity label="Humidity [%.1f %%]"

This returns "Humidity     5600 %".

Similar problem if humidity is defined as following on Sitemap:

> Text item=LR_Humidity label="Humidity"
This returns "Humidity     56.1 **one**".

**Comment by Rossko57** [comment]
> This returns "Humidity 56.1 one".

I think this reveals the nature of the issue.  The 56.1 has been stored in the Number:Dimensionless Item in default units.  So far as openHAB knows, it represents a **ratio** of 1:56.1, spelled out "one to fifty-six".
If we then ask for that to be displayed in percent, 5600% is a reasonable interpretation.  A conversion has effectively been applied from the stored ratio to the requested percentage format.

It does seem to me that the %.1f should really be displaying exactly **5610.0 %** so maybe that needs a closer look.

I wonder if the original writer's surprise comes from updating Number:Dimensionless (maybe by binding or transform) with a blank or null UoM unit, where he should specify percent e.g.  **56.1 | %**

**Comment by Rossko57** [comment]
I experimented in OH 2.4, but was not able to reproduce the original post.
item -
`Number:Dimensionless testNMD "test item [%.1f %%]"`
sitemap -
```
		Text item=testNMD
		Text item=testNMD label="show default [%.1f %unit%]"
		Text item=testNMD label="force percent [%.1f %%]"
		Text item=testNMD label="force decibel [%.1f dB]"
		Text item=testNMD label="force ratio [%.1f one]"
```

Run `testNMD.postUpdate(56.1)`
display
> test item 56.1 %
> show default 56.1 %
> force percent 56.1 %
> force decibel -2.5 dB
> force ratio 0.6 one

What has happened here is that posting unit-less value "56.1" has assumed a default unit from the unit specified in the Item's [format]

For comparison, `testNMD.postUpdate(56.1 | "dB")`
display
> test item 40738027.8 %
> show default 56.1 dB
> force percent 40738027.8 %
> force decibel 56.1 dB
> force ratio 407380.3 one
which is what I expected.  Note the raw Item display using its default label has converted to %, but the default Item unit is now changed to dB because we postupdated that.

The closest I can get to the original post is by removing the units clue in the Item
`Number:Dimensionless testNMD "test item [%.1f]"`
and updating with no unit, `testNMD.postUpdate(56.1)`
display
> test item 56.1
> show default 56.1 one
> force percent 5610.0 %
> force decibel 17.5 dB
> force ratio 56.1 one
With no units clue in the Item or update, OH has fallen back on the system default for Number:Dimensionless and assumed 'one' or ratio type units.

This all seems to be working as intended to me.
